src/debuggability.js

```diff
@@ -178,6 +188,79 @@ Promise.prototype._propagateFrom = function (parent, flags) {
     USE(flags);
 };
 
+Promise.enableMonitoring = function () {
+    function registerPromise() {
+        if (Promise.monitor) {
+            Promise.monitor._promiseIdCounter++;
+            if (Promise.monitor._promiseIdCounter === Number.MAX_VALUE) {
```

Comment:
This check should be removed, there is no way this is never happening:

Firstly, because an int cannot be incremented with `++` after  `9007199254740992`:

```js
var a = 9007199254740992;
var l = 1000; while(l--) a++`
// 9007199254740992
a;
```

Secondly, if you create a **million** promises per **second**, it would take **285 years** to reach **9007199254740992**.

So in practice these conditions will never be executed and they are therefore dead code.